### DISCUSSION TITLE
Importing SSS models using pyAEDT

### DISCUSSION THREAD

**Discussion by mikevacany** [ansys/pyaedt#4975] [Q&A] [answered]
Hi all,

is it possible to import into circuit SSS model in a similar way as Touchstone models are imported using create_touchstone_component() or spice component using create_component_from_spicemodel()?

If json/IronPython API is only choice, is there an example how to create component with .pins list attribute, so .pins[x].connect_to_component() function can be used to interconnect components?

Thanks in advance,
Michael

**Comment by gkorompi** [comment]
Hello @mikevacany. We are currently implementing a function to support .sss files. FYI https://github.com/ansys/pyaedt/issues/5804

**Reply by mikevacany** [reply]
Thank you for update, @gkorompi! I tried to import my .sss model using create_nexxim_state_space_component(), but the Schematic symbol ports are named with numbers and not with signal names. Is this intentional limitation or I am doing something wrong? If I import the same file in GUI, I get signal names on the .sss instance ports.

**Reply by gkorompi** [reply, marked as answer]
Hello @mikevacany, we actually implemented a port_names input argument in order to support this difference:

https://aedt.docs.pyansys.com/version/stable/API/_autosummary/ansys.aedt.core.modeler.circuits.primitives_nexxim.NexximComponents.create_model_from_nexxim_state_space.html#create_model_from_nexxim_state_space

The way .sss files are written are in bit form, hence we cannot access the port names, instead for all cases we support the default naming (with numbers), and in other cases you can input the port name list. 

Let me know if this is limiting your effort. 

Georgios

**Reply by mikevacany** [reply]
Hello @gkorompi,

thank you for fast response. I already parsed the Touchstone file using create_touchstone_component(), so I would expect compatible behavior and API. Keeping the compatible behavior and API is something which should be followed in general.

For me alternative to .SSS file import is also Touchstone 2.0 import, but this is not supported, right? 

I will then go with SSS and the list of signal names.

Thank you,
Michael
